### ISSUE TITLE
Announcement: Moving minimum VS support to 2017 and msbuild 15 

### ISSUE THREAD

**Issue by weshaggard** [dotnet/corefx#24320]
For the core repo's (standard, corefx, coreclr, core-setup) we are planning to stop supporting VS2015 and msbuild 14 and raising the minimum version to VS2017 and msbuild 15. 

The reason for this is that there are a number of new features in VS 2017 in the compiler and msbuild that we want to start taking advantage of. Before we make that move we wanted to let folks know and give everyone a chance to speak up if there are any issues with moving.

@dotnet/dnceng note that this will require updating our CI and official build machines. Please let me know what needs to happen for that to proceed.

@karelz what set of alias' would be best to ping on this issue to raise awareness?

**Comment by karelz** [comment]
I think the biggest impact is external. Internal people are on VS 2017 (or should be).

Maybe Xamarin might want to know: @marek-safar @migueldeicaza 
Linux-focused partners are not impacted (Red Hat, Samsung).
@danmosemsft anyone else?

It is also worth mentioning that @weshaggard is trying to remove our dependency on VS. It still may take a few months, but we are heading in the right direction (I can't find the GH issue if there is one for it).

**Comment by marek-safar** [comment]
@karelz that's fine by us

**Comment by 4creators** [comment]
Is there a chance for an update to new csproj/proj format as well (genaral dogfooding idea)?

**Comment by weshaggard** [comment]
@4creators we do plan to move to SDK style csproj's overtime but it will not be as part of this transition. We will allow the use of some of the new msbuild features like providing metadata as attributes, but there won't be a large sweep to do that right now.

**Comment by weshaggard** [comment]
@dotnet/dnceng we are still finding some machines without VS2017. Are we ready to pull the trigger and move forward and make our scripts fail on any machines without VS2017? If we do that I expect there might be an influx of first responder issues coming in so I want to coordinate that effort.

**Comment by mmitche** [comment]
@weshaggard Not yet.  There are still some that need cleanup, and official build machines have 2017, but there aren't enough of them yet to satisfy load.  @ilyas is working on this.

**Comment by tarekgh** [comment]
@mmitche @weshaggard do we still have something to do here? or this is done?

**Comment by weshaggard** [comment]
We still need to go through the repo's and remove the VS2015 support.

**Comment by tarekgh** [comment]
@weshaggard do we still targeting 2.1 for that? are you the one doing this work?

**Comment by weshaggard** [comment]
It isn't critical for 2.1 and I doubt we will get it cleaned up by then.

**Comment by weshaggard** [comment]
As part of cleaning up the VS15 work we should update our VSConfigurationGenerator to match what VS2017 outputs by default. See https://github.com/dotnet/corefx/pull/27831#discussion_r173226079.

**Comment by kasper3** [comment]
@weshaggard, isn't #27831 exactly the opposite of moving to VS2017 or is the plan to reconcile the versions to 2015 first, then write a tool that will update solution and project files together?

**Comment by weshaggard** [comment]
@kasper3 #27831 is just about getting our solutions updated based on our current generation tool. Once we work on cleaning out the VS2015 we will update the generation tool to also update all the solutions.

**Comment by weshaggard** [comment]
I'd like to still keep this issue open until we clean-up the various VS2015 support things in various scripts in our repo (see https://github.com/dotnet/corefx/search?q=vs2015&unscoped_q=vs2015).

**Comment by weshaggard** [comment]
@ViktorHofer I agree we fully support VS2017 but there are still plenty of places where we reference and do extra work to try and support VS2015 which I still believe we should clean-up and remove.

**Comment by ViktorHofer** [comment]
ok, reopening 😁

**Comment by ViktorHofer** [comment]
@ericstj do you know which places are left that check for VS2015?

**Comment by mmitche** [comment]
IIRC core-setup is still building on dev14 officially...

**Comment by ViktorHofer** [comment]
I was mainly talking about corefx but good to know.

**Comment by weshaggard** [comment]
@ViktorHofer see https://github.com/dotnet/corefx/search?q=vs2015&unscoped_q=vs2015

**Comment by am11** [comment]
With a bit more curious query, it would be: https://github.com/dotnet/corefx/search?q="studio+2015"+OR+"vs2015"+OR+"vs+2015"+OR+"visualstudio2015" :)
